themes/src/main/resources/theme/keycloak-preview/account/resources/app/content/my-resources-page/EditTheResource.tsx

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as React from 'react';
+
+import {
+    Button,
+    Modal,
+    DataList,
+    DataListItemRow,
+    DataListItemCells,
+    DataListCell,
+    DataListItem,
+    ChipGroup,
+    ChipGroupToolbarItem,
+    Chip
+} from '@patternfly/react-core';
+
+import { EditAltIcon } from '@patternfly/react-icons';
+
+import { Resource, Permission, Scope } from './MyResourcesPage';
+import { Msg } from '../../widgets/Msg';
+import { AccountServiceClient } from '../../account-service/account.service';
+import { ContentAlert } from '../ContentAlert';
+
+interface EditTheResourceProps {
+    resource: Resource;
+    permissions: Permission[];
+    onClose: (resource: Resource, row: number) => void;
+    row: number;
+}
+
+interface EditTheResourceState {
+    isOpen: boolean;
+}
+
+export class EditTheResource extends React.Component<EditTheResourceProps, EditTheResourceState> {
+    protected static defaultProps = { permissions: [], row: 0 };
+
+    public constructor(props: EditTheResourceProps) {
+        super(props);
+
+        this.state = {
+            isOpen: false,
+        };
+    }
+
+    private clearState(): void {
+        this.setState({
+        });
+    }
+
+    private handleToggleDialog = () => {
+        if (this.state.isOpen) {
+            this.setState({ isOpen: false });
+        } else {
+            this.clearState();
+            this.setState({ isOpen: true });
+        }
+    };
+
+    async deletePermission(permission: Permission, scope: Scope): Promise<void> {
+        permission.scopes.splice(permission.scopes.indexOf(scope), 1);
+        await AccountServiceClient.Instance.doPut(`/resources/${this.props.resource._id}/permissions`, {data: [permission]});
+        ContentAlert.success(Msg.localize('shareSuccess'));
+        this.props.onClose(this.props.resource, this.props.row);
+    }
+
+    public render(): React.ReactNode {
+        return (
+            <React.Fragment>
+                <Button variant="link" onClick={this.handleToggleDialog}>
+                    <EditAltIcon /> Edit
+                </Button>
+
+                <Modal
+                    title={'Edit the resource - ' + this.props.resource.name}
+                    isLarge={true}
+                    width={'45%'}
```

Comment:
@edewit Just wondering.  Have you tried this on small devices like cell phones?  How does it look?

Reply:
that is a very stupid setting both seems that it's a copy paste, removed it in both places. Without it it looks like this:
![image](https://user-images.githubusercontent.com/51133/79193153-5780b380-7e2a-11ea-83ec-e3b33e626341.png)

Reply:
on a wide screen without the 45% it's a bit wide, but still okay in my opinion:
![image](https://user-images.githubusercontent.com/51133/79196264-0a074500-7e30-11ea-9861-c5746c643e13.png)